mappings/net/minecraft/world/gen/GeneratorOptions.mapping

```diff
@@ -0,0 +1,103 @@
+	METHOD method_28009 getDefaultOptions ()Lnet/minecraft/class_5285;
+	METHOD method_28011 getEndGenerator (J)Lnet/minecraft/class_2794;
+		ARG 0 seed
+	METHOD method_28012 withBuffetType (Lnet/minecraft/class_5285$class_5286;Ljava/util/Set;)Lnet/minecraft/class_5285;
+		ARG 2 biomes
+	METHOD method_28014 withGenerator (Lnet/minecraft/class_5285$class_5287;Lcom/mojang/datafixers/Dynamic;Lnet/minecraft/class_2794;)Lnet/minecraft/class_5285;
+		ARG 2 generatorSettings
+	METHOD method_28015 withMapType (Lnet/minecraft/class_5285$class_5288;)Lnet/minecraft/class_5285;
+	METHOD method_28017 getBiomeSource (Lcom/mojang/datafixers/DynamicLike;J)Lnet/minecraft/class_1966;
+		ARG 1 seed
+	METHOD method_28018 getRegistryEntry (Lcom/mojang/datafixers/DynamicLike;Lnet/minecraft/class_2378;Ljava/lang/Object;)Ljava/lang/Object;
+		ARG 0 id
+		ARG 1 registry
+		ARG 2 fallback
+	METHOD method_28019 withFlatGenerator (Lnet/minecraft/class_3232;)Lnet/minecraft/class_5285;
+	METHOD method_28021 fromProperties (Ljava/util/Properties;)Lnet/minecraft/class_5285;
+	METHOD method_28023 fromTag (Lnet/minecraft/class_2487;Lcom/mojang/datafixers/DataFixer;I)Lnet/minecraft/class_5285;
+		ARG 0 tag
+		ARG 1 dataFixer
+		ARG 2 version
+	METHOD method_28024 withHardcore (ZLjava/util/OptionalLong;)Lnet/minecraft/class_5285;
+		ARG 1 hardcore
+		ARG 2 seed
+	METHOD method_28025 toTag ()Lnet/minecraft/class_2487;
+	METHOD method_28026 getNetherGenerator (J)Lnet/minecraft/class_2794;
+		ARG 0 seed
+	METHOD method_28027 getBuffetWorldSettings (Lnet/minecraft/class_5285$class_5286;Ljava/util/Set;)Lcom/mojang/datafixers/Dynamic;
+		ARG 0 worldGeneratorType
+		ARG 1 biomes
+	METHOD method_28028 getSeed ()J
+	METHOD method_28029 shouldGenerateStructures ()Z
+	METHOD method_28030 hasBonusChest ()Z
+	METHOD method_28031 getChunkGenerators ()Ljava/util/Map;
+	METHOD method_28032 getChunkGenerator ()Lnet/minecraft/class_2794;
+	METHOD method_28033 isDebugWorld ()Z
+	METHOD method_28034 isFlatWorld ()Z
+	METHOD method_28035 isLegacyCustomizedType ()Z
+	METHOD method_28036 withBonusChest ()Lnet/minecraft/class_5285;
+	METHOD method_28037 toggleGenerateStructures ()Lnet/minecraft/class_5285;
+	METHOD method_28038 toggleBonusChest ()Lnet/minecraft/class_5285;
+	METHOD method_28039 getWorldType ()Lnet/minecraft/class_5285$class_5288;
+	METHOD method_28040 getFlatChunkGeneratorConfig ()Lnet/minecraft/class_3232;
+	METHOD method_28041 getBuffetConfig ()Lcom/mojang/datafixers/util/Pair;
+	METHOD method_28042 getLevelToMapTypeMap ()Ljava/util/Map;
+	CLASS class_5286 WorldGeneratorType
+		FIELD field_24539 BY_ID Ljava/util/Map;
+		FIELD field_24540 id Ljava/lang/String;
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+			ARG 3 id
+		METHOD method_28043 getText ()Lnet/minecraft/class_2561;
+		METHOD method_28045 byId (Ljava/lang/String;)Lnet/minecraft/class_5285$class_5286;
+			ARG 0 id
+		METHOD method_28046 getId ()Ljava/lang/String;
+	CLASS class_5287 LevelType
```

Comment:
How is level type different from world generator type etc?

Reply:
We spent like half an hour deciding what to name these three classes. This is called LevelType, because it maps directly to the level-type field in server.properties. MapType is called that because that's the localization string of the button that controls that setting (although it was still unclear to us how that button showed options that are not in the MapType enum but are in LevelType like Large Biomes. WorldGeneratorType is used in the buffet generator. Maybe BuffetWorldGeneratorType is actually a better name for that one.

Reply:
That client button appears to me as a fancy layer above the actual level type (this boi) that is used in actual save. Mojang does offer some sort of shortcut for powerful maneuvers on client